### PR TITLE
Fix beam.Row.__eq__ to ignore key order

### LINKED ISSUE #32832: [Failing Test]: PreCommit Yaml Xlang Direct is broken
### What happened?

Tests are [failing](https://github.com/apache/beam/actions/workflows/beam_PreCommit_Yaml_Xlang_Direct.yml?query=branch%3Amaster+is%3Acompleted) after #32757 was merged

Failing tests are:
- `apache_beam/yaml/integration_tests.py::MapTest::test_Filter-generic_InlineProvider_MapToFields-generic_ExternalJavaProvider_1`
- `apache_beam/yaml/integration_tests.py::MapTest::test_Filter-generic_ExternalJavaProvider_MapToFields-generic_ExternalJavaProvider_3`

Error (reformatted for readability):

```
FAILED apache_beam/yaml/integration_tests.py::MapTest::test_Filter-generic_InlineProvider_MapToFields-generic_ExternalJavaProvider_1 -
apache_beam.testing.util.BeamAssertException: Failed assert: 
[Row(element=100, named_field=100, literal_int=10, literal_float=1.5, literal_str='abc'), Row(element=200, named_field=200, literal_int=10, literal_float=1.5, literal_str='abc')] 
== 
[Row(element=100, literal_int=10, named_field=100, literal_float=1.5, literal_str='abc'), Row(element=200, literal_int=10, named_field=200, literal_float=1.5, literal_str='abc')], 
unexpected elements 
[Row(element=100, literal_int=10, named_field=100, literal_float=1.5, literal_str='abc'), Row(element=200, literal_int=10, named_field=200, literal_float=1.5, literal_str='abc')], 
missing elements 
[Row(element=100, named_field=100, literal_int=10, literal_float=1.5, literal_str='abc'), Row(element=200, named_field=200, literal_int=10, literal_float=1.5, literal_str='abc')] 
[while running 'AssertEqual/assert_that/Match']
```

### Issue Failure

Failure: Test is continually failing

### Issue Priority

Priority: 1 (unhealthy code / failing or flaky postcommit so we cannot be sure the product is healthy)

### Issue Components

- [ ] Component: Python SDK
- [X] Component: Java SDK
- [ ] Component: Go SDK
- [ ] Component: Typescript SDK
- [ ] Component: IO connector
- [X] Component: Beam YAML
- [ ] Component: Beam examples
- [ ] Component: Beam playground
- [ ] Component: Beam katas
- [ ] Component: Website
- [ ] Component: Infrastructure
- [ ] Component: Spark Runner
- [ ] Component: Flink Runner
- [ ] Component: Samza Runner
- [ ] Component: Twister2 Runner
- [ ] Component: Hazelcast Jet Runner
- [ ] Component: Google Cloud Dataflow Runner

### PR DESCRIPTION
**Pull request by Polber**
Beam Rows in the Python SDK are compared using an `__eq__` method that considers the position of each key in the row. The case should be that order does not matter so long as all keys exists and have equal values of equal types.

Fixes #32832

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://github.com/apache/beam/blob/master/CONTRIBUTING.md#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI or the [workflows README](https://github.com/apache/beam/blob/master/.github/workflows/README.md) to see a list of phrases to trigger workflows.

### CODE REVIEW

File: sdks/python/apache_beam/pvalue.py [hunk 1]
```diff
@@ -682,7 +682,8 @@ def __eq__(self, other):
         type(self) == type(other) and
         len(self.__dict__) == len(other.__dict__) and all(
             s == o for s,
-            o in zip(self.__dict__.items(), other.__dict__.items())))
+            o in zip(
```
Comment (robertwb) [inline sdks/python/apache_beam/pvalue.py:685]: Write `for (s, o) in zip(...)` for better line breaks.

**Comment by robertwb** [comment]
I'm not positive we want comparison to be order-independent, see e.g. the comments on https://github.com/apache/beam/pull/23876

**Comment by Polber** [comment]
> I'm not positive we want comparison to be order-independent, see e.g. the comments on #23876

Yeah it appears I totally missed a crucial comment
```
Note that in Beam 2.30.0 and later, Row objects are sensitive to field order.
So `Row(x=3, y=4)` is not considered equal to `Row(y=4, x=3)`.
```
It seems odd to me that order should matter on a schema'd data type, but I presume there is a use-case I'm not considering. In any case, I opened #32851 to fix the failing YAML tests

**Comment by Polber** [comment]
Closing in favor of #32851

**Pull request closed without merging**